bin/release.py

```diff
@@ -0,0 +1,124 @@
+"""A script which triggers and wait for an azure devops to complete a release"""
+from ast import parse
+import os
+import sys
+import time
+from azure.devops.connection import Connection
+from msrest.authentication import BasicAuthentication
+
+FAILED_STATUSES = ["canceled", "partiallySucceeded", "rejected"]
+COMPLETED_STATUSES = ["succeeded", "canceled", "partiallySucceeded", "rejected"]
+
+
+def create_ado_connection(organization, ADO_PAT) -> Connection:
+    """Returns an ADO connection to call the ADO Rest APIs."""
+
+    organization_url = "https://dev.azure.com/{0}".format(organization)
+    credentials = BasicAuthentication("", ADO_PAT)
+    connection = Connection(base_url=organization_url, creds=credentials)
+
+    return connection
+
+
+def get_release_definition(project, pipeline_name, release_client):
+    """Returns the ADO definition for the given pipeline"""
+    # Get all release definitions in a given project
+    project_release_definitions = release_client.get_release_definitions(project)
+
+    # Filter release definitions with the given pipeline name
+    pipeline_release_definitions = []
+    for definition in project_release_definitions.value:
+        if definition.name == pipeline_name:
+            pipeline_release_definitions.append(definition)
+
+    if pipeline_release_definitions is None or len(pipeline_release_definitions) == 0:
+        error_message = f"Pipeline named {pipeline_name} not found in project {project}"
+        raise Exception(error_message)
+
+    if len(pipeline_release_definitions) > 1:
+        error_message = (
+            f"More than 1 Pipeline named {pipeline_name} found in project {project}"
+        )
+        raise Exception(error_message)

```

Comment:
Shouldn't this be a separate error message than the one above? If we have more than one pipeline names, then returning an error message saying none found is confusing.

Reply:
Yes Shalini! I didn't get you. I created two different error messages for both scenarios.

Reply:
Isn't it a separate message? In both cases we `raise` an exception causing an early exit. This error will never be hit if the other one is, so no one will see both. We should have exactly one pipeline with this name, so `0` and `>1` are both different errors.